src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -206,6 +214,22 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             Return OperationKind.AssignmentExpression
         End Function
+
+        Public Overrides Sub Accept(visitor As IOperationVisitor)
+            If Me.ExpressionKind() = OperationKind.CompoundAssignmentExpression Then
+                visitor.VisitCompoundAssignmentExpression(Me)
+            Else
+                visitor.VisitAssignmentExpression(Me)
+            End If
+        End Sub
```

Comment:
This is a tough invariant to preserve without doing some rearrangement of the bound nodes classes.